src/amplitude-client.js

```diff
@@ -739,18 +739,22 @@ AmplitudeClient.prototype._saveReferrer = function _saveReferrer(referrer) {
  */
 AmplitudeClient.prototype.saveEvents = function saveEvents() {
   try {
+    const unsentEvents = JSON.stringify(this._unsentEvents.map(({event}) => event));
+
     if (AsyncStorage) {
-      AsyncStorage.setItem(this.options.unsentKey + this._storageSuffix, JSON.stringify(this._unsentEvents));
+      AsyncStorage.setItem(this.options.unsentKey + this._storageSuffix, unsentEvents);
     } else {
-      this._setInStorage(localStorage, this.options.unsentKey, JSON.stringify(this._unsentEvents));
+      this._setInStorage(localStorage, this.options.unsentKey, unsentEvents);
     }
   } catch (e) {}
 
   try {
+    const unsentIdentifys = JSON.stringify(this._unsentIdentifys.map(unsentIdentify => unsentIdentify.event));
+
     if (AsyncStorage) {
-      AsyncStorage.setItem(this.options.unsentIdentifyKey + this._storageSuffix, JSON.stringify(this._unsentIdentifys));
+      AsyncStorage.setItem(this.options.unsentIdentifyKey + this._storageSuffix, unsentIdentifys);
     } else {
-      this._setInStorage(localStorage, this.options.unsentIdentifyKey, JSON.stringify(this._unsentIdentifys));
+      this._setInStorage(localStorage, this.options.unsentIdentifyKey, unsentIdentifys);
```

Comment:
hmmm.. i worry that someone could be skimming this code and easily overlook that `this. _setInStorage` expects string and we're running `JSON.stringify()` above. Would it make sense to move the `JSON.stringify()` logic to `this. _setInStorage`?